src/core/batch/batch.ts

```diff
@@ -20,47 +23,73 @@ interface BloomFilterData {
   RowGroupIndex: number;
 }
 
+interface BatchFileData {
+  url: URL;
+  hash: HexString;
+}
+
+type BatchMessage<T extends DSNPType> = AsyncOrSyncIterable<DSNPMessageSigned<DSNPTypedMessage<T>>>;
```

Comment:
We might need to pluralize this at least. It reads oddly down below when we replaced `DSNPMessage[]` with `BatchMessage<T>`

Perhaps `BatchMessagesIterable` event?